mat/format.go

```diff
@@ -193,6 +213,252 @@ func format(m Matrix, prefix string, margin int, dot byte, squeeze bool, fs fmt.
 	}
 }
 
+// formatMATLAB prints a MATLAB representation of m to the fs io.Writer. The format character c
+// specifies the numerical representation of elements; valid values are those for float64
+// specified in the fmt package, with their associated flags.
+// The printed range of the matrix can be limited by specifying a positive value for margin;
+// If squeeze is true, column widths are determined on a per-column basis.
+//
+// formatMATLAB will not provide Go syntax output.
+func formatMATLAB(m Matrix, prefix string, _ int, _ byte, squeeze bool, fs fmt.State, c rune) {
+	rows, cols := m.Dims()
+
+	prec, pOk := fs.Precision()
+	width, _ := fs.Width()
+	if !fs.Flag('#') {
+		format := fmtString(fs, c, prec, width)
+		fs.Write([]byte{'['})
```

Comment:
just a general comment, but perhaps all these little `Write([]byte)` could be accumulated into a local `[]byte` (or `bufio.Writer`) and flushed at every row in the `#` case.

Reply:
I'll have a look at doing this.

Reply:
We don't know what the underlying `io.Writer` is, so I don't think this is worthwhile. It may already be buffering, and if it is not, that doesn't matter either.